src/components/table/index.tsx

```diff
@@ -167,6 +169,16 @@ export default function EnhancedTable<T extends TableRowType>({
     }
   }, [rows, setSelectedRows]);
 
+  useEffect(() => {
+    // this is not most elegant but we want to do this if table was sorted by some column in a presorted table
+    // but we don't know when the data changes, hence this useEffect on the data size
+    if (rows.length && isPresorted) {
+      // we want to set page back to 1 if the data changes on presorted lists,
+      // this is because the data was reset due to some sort behavior refetching new data
+      handleChangePage({}, 1);
```

Comment:
Don't we need to execute this for any row change? (Not only a length change)

Reply:
ugh good point, the number of rows may stay the same...

Reply:
updated, thanks for catching this..